File: Python3/Medium/Accepted/corporate-flight-bookings_6.py
class Solution:
    def corpFlightBookings(self, bookings: List[List[int]], n: int) -> List[int]:
        
        l = [0] * (n+1)
        for b in bookings:
            l[b[0]-1] += b[2]
            if b[1] < n:
                l[b[1]] -= b[2]
        for i in range(1, n):
            l[i] += l[i-1]
        return l[:-1]

File: Rust/Medium/Accepted/corporate-flight-bookings_6.rs
impl Solution {
    pub fn corp_flight_bookings(bookings: Vec<Vec<i32>>, n: i32) -> Vec<i32> {
        
        let mut ans = vec![0; n as usize];
        for booking in bookings {
            let (first, last, seats) = (booking[0] as usize - 1, booking[1] as usize - 1, booking[2]);
            ans[first] += seats;
            if last + 1 < n as usize {
                ans[last + 1] -= seats;
            }
        }
        for i in 1..n as usize {
            ans[i] += ans[i - 1];
        }
        ans
    }
}